kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/server/GwtWireServiceImpl.java

```diff
@@ -495,19 +495,21 @@ public GwtWiresConfiguration updateWireConfiguration(final GwtXSRFToken xsrfToke
                     Filter filter = bundleContext.createFilter(filterString);
                     final ServiceTracker producerTracker = new ServiceTracker(bundleContext, filter, null);
                     producerTracker.open();
-                    producerTracker.waitForService(5000);
-                    producerTracker.close();
 
                     // track and wait for the receiver
                     final String cPid = wireHelperService.getServicePid(receiverPid);
                     filterString = "(" + Constants.SERVICE_PID + "=" + cPid + ")";
                     filter = bundleContext.createFilter(filterString);
                     final ServiceTracker consumerTracker = new ServiceTracker(bundleContext, filter, null);
                     consumerTracker.open();
-                    consumerTracker.waitForService(5000);
-                    consumerTracker.close();
+
+                    producerTracker.waitForService(SERVICE_WAIT_TIMEOUT);
+                    consumerTracker.waitForService(SERVICE_WAIT_TIMEOUT);
 
                     wireService.createWireConfiguration(emitterPid, receiverPid);
+
+                    producerTracker.close();
```

Comment:
Move it before creating wire configuration because if the wire configuration creation fails, the tracker will never be closed.